backend/src/main/kotlin/com/infowings/catalog/data/guid/GuidService.kt

```diff
@@ -0,0 +1,101 @@
+package com.infowings.catalog.data.guid
+
+import com.infowings.catalog.common.*
+import com.infowings.catalog.common.objekt.PropertyUpdateResponse
+import com.infowings.catalog.common.objekt.Reference
+import com.infowings.catalog.data.Subject
+import com.infowings.catalog.data.aspect.toAspectPropertyVertex
+import com.infowings.catalog.data.aspect.toAspectVertex
+import com.infowings.catalog.data.objekt.*
+import com.infowings.catalog.data.reference.book.toReferenceBookItemVertex
+import com.infowings.catalog.data.subject.toSubject
+import com.infowings.catalog.data.subject.toSubjectVertex
+import com.infowings.catalog.storage.*
+import com.orientechnologies.orient.core.record.ODirection
+
+data class EntityMetadata(val guid: String, val entityClass: String, val id: String)
+
+class GuidService(private val db: OrientDatabase, private val dao: GuidDaoService) {
+    fun metadata(guids: List<String>): List<EntityMetadata> {
+        return transaction(db) {
+            dao.find(guids).map { guidVertex ->
+                val vertex = dao.vertex(guidVertex)
+                EntityMetadata(guid = guidVertex.guid, entityClass = vertex.vertexClass(), id = vertex.id)
+            }
+        }
+    }
+
+    fun findAspects(guids: List<String>): List<AspectData> {
+        return transaction(db) {
+            dao.find(guids).map { guidVertex ->
```

Comment:
Both variants are O(n).
Yes, mapNotNull assumes sligtly better coefficient but it is less idiomatic.
It is matter of personal preference.
I like chain of map/filter more because it is lingua franca of functional programming style.
If it becomes bottleneck (I guess code dealing with Orient will be one though) guids shoud be put into some kind lazy wrapper and it will result in performance like one of mapNotNull.